### PR TITLE
ts-node: add page

### PR DESCRIPTION
**Pull request by SpiderMath**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

I have checked all of stuff and I follow all the guidelines as far as I know.
I am adding TS-Node which is a really useful CLI tool with 8 million+ Downloads on NPM (https://www.npmjs.com/package/ts-node), since it was not listed in the Repo and but is used by a ton of people.

### CODE REVIEW

File: pages/common/ts-node.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# ts-node
+
+> Runs TypeScript code directly, without any compiling.
```
Comment (marchersimon) [inline pages/common/ts-node.md:3]:
```suggestion
> Run TypeScript code directly, without any compiling.
```

File: pages/common/ts-node.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+> More information at: https://www.npmjs.com/package/ts-node
```
Comment (marchersimon) [inline pages/common/ts-node.md:4]:
```suggestion
> More information: <https://www.npmjs.com/package/ts-node>.
```

File: pages/common/ts-node.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+- Execute a TS File `foobar.ts` without compiling ('node' + 'tsc')
+
+`ts-node {{foobar.ts}}`
```
Comment (marchersimon) [inline pages/common/ts-node.md:8]:
```suggestion
- Execute a TS file without compiling (`node` + `tsc`).

`ts-node {{path/to/file.ts}}`
```

File: pages/common/ts-node.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+- TS-Node Loads tsconfig.json by default. To prevent that
+
+`ts-node --skip-project {{foobar.ts}}`
```
Comment (marchersimon) [inline pages/common/ts-node.md:12]:
```suggestion
- Execute a TS file without loading `tsconfig.json`:

`ts-node --skip-project {{path/to/file.ts}}`
```

File: pages/common/ts-node.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+- Loads TS-Node Help
```
Comment (marchersimon) [inline pages/common/ts-node.md:14]:
```suggestion
- Display TS-Node help:
```
Also, the `--help` example (if any) usually goes to the bottom.

File: pages/common/ts-node.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+`ts-node --help`
+
+- Eval TS Code directly on the terminal
+
+`ts-node -e 'console.log("Hello World")'`
```
Comment (marchersimon) [inline pages/common/ts-node.md:20]:
```suggestion
- Evaluate TS code directly on the terminal:

`ts-node -e '{{console.log("Hello World")}}'`
```

File: pages/common/ts-node.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+- Run code in Script Mode for `foobar.ts`
+
+`ts-node --script-mode {{foobar.ts}}`
```
Comment (marchersimon) [inline pages/common/ts-node.md:24]:
```suggestion
- Execute a TS file in script mode:

`ts-node --script-mode {{path/to/file.ts}}`
```

File: pages/common/ts-node.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+- Only transpile the TS File `foobar.ts` to JS
+
+`ts-node --transpile-only {{foobar.ts}}`
```
Comment (marchersimon) [inline pages/common/ts-node.md:28]:
```suggestion
- Transpile a TS file to JS without executing it:

`ts-node --transpile-only {{path/to/file.ts}}`
```

**Review by marchersimon — CHANGES_REQUESTED** [review]
Looks good overal, just a few tldr conventions:

File: pages/common/ts-node.md (hunk 6 above)
Comment (bl-ue) [reply pages/common/ts-node.md:20]: I'd also like it to be "Evaluate TypeScript code ..."

Comment (marchersimon) [reply pages/common/ts-node.md:20]: Should we generally replace `TS` with `TypeScript`? The descriptions aren't really long so that wouldn't be a problem.

**Comment by bl-ue** [comment]
Sorry, I forgot to wait before running the checks 😖

File: pages/common/ts-node.md (hunk 6 above)
Comment (bl-ue) [reply pages/common/ts-node.md:20]: Yes, I think so.  I would have waited to send that comment but I didn't notice that we were using `TS` all over before that... 🤣

File: pages/common/ts-node.md (hunk 8 above)
Comment (SpiderMath) [reply pages/common/ts-node.md]: That's a lotta suggestions but Alright lol

**Comment by SpiderMath** [comment]
I have fixed most of the stuff and made TS to TypeScript. I'm just so used to calling TypeScript TS that I forgot that I was using an abbreviation :sweat_smile:

**Comment by bl-ue** [comment]
@SpiderMath if you're not sure what I meant by 'running the checks', see [this recently published GitHub blog article](https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/) tha explains that stuff ;)

File: pages/common/ts-node.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+- Execute a TypeScript file without loading `tsconfig.json`:
+
+`ts-node --skip-project {{path/to/file.ts}}`
+
+- Display TS-Node help:
+
+`ts-node --help`
+
+- Evaluate TypeScript code directly on the terminal:
```
Comment (bl-ue) [inline pages/common/ts-node.md:18]: Perhaps this will read better?
```suggestion
- Evaluate TypeScript code passed as a literal on the command-line:
```

Comment (navarroaxel) [reply pages/common/ts-node.md:18]: @bl-ue Hunspell prefers `command line`.

Comment (bl-ue) [reply pages/common/ts-node.md:18]: Well, Hunspell doesn't necessarily need to be honored always.  In fact, there are a great many words in all of the English pages that Hunspell doesn't like, but are, in fact, valid.

#5848

File: pages/common/ts-node.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+`ts-node {{path/to/file.ts}}`
+
+- Execute a TypeScript file without loading `tsconfig.json`:
+
+`ts-node --skip-project {{path/to/file.ts}}`
+
+- Display TS-Node help:
+
+`ts-node --help`
```
Comment (peterbabic) [inline pages/common/ts-node.md:16]: Is this a helpful example? I have seen some comments that a help command, if included, is generally present as a last example.

Comment (SpiderMath) [reply pages/common/ts-node.md]: I have thus put it as the last example

**Comment by SpiderMath** [comment]
Is it alright now?

File: pages/common/ts-node.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
+# ts-node
+
+> Run TypeScript code directly, without any compiling.
+> More information: <https://www.npmjs.com/package/ts-node>.
+
+- Execute a TypeScript file without compiling (`node` + `tsc`).
```
Comment (peterbabic) [inline pages/common/ts-node.md:6]:
```suggestion
- Execute a TypeScript file without compiling (`node` + `tsc`):
```

**Review by peterbabic — CHANGES_REQUESTED** [review]

File: pages/common/ts-node.md (hunk 11 above)
Comment (peterbabic) [reply pages/common/ts-node.md:6]: This is what the bot is complaining about @SpiderMath 

Can you please fix it?

**Comment by SpiderMath** [comment]
Damn, I didn't notice that. My bad :sweat:

**Review by bl-ue — APPROVED** [review]

**Review by marchersimon — APPROVED** [review]
Thanks @SpiderMath

**Pull request merged**